### PR TITLE
Fix unit tests and general refactor

### PR DESCRIPTION
**Pull request by ebrahim-soliman**
1. Fix all unit tests(except one).
2. Add unit tests to TestSweetsConfigFileService.
3. Refactor testsweets file to reduce boilerplate.
4. Remove duplicate code especially error messages.
5. Change weird makeinstance way of polymorphism to normal and refactor where exists.
6. update readme.

### CODE REVIEW

File: packages/testsweets/test/test_sweets_config_file_service_test.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    });
+    test(
+        "Should split any list of strings by the '=' sign to List<MapEntry<String, String>>",
+        () async {
+      TestSweetsConfigUtils testSweetsConfigUtils = TestSweetsConfigUtils();
+      var result = testSweetsConfigUtils
+          .splittingStringOnEqualSign(testsweetFileContent);
+      expect(result[0].value, '3OezzTovG9xFTE5Xw2w1');
+    });
+    test(
```
Comment (FilledStacks) [inline packages/testsweets/test/test_sweets_config_file_service_test.dart:38]: This test is the same as the one above it

File: packages/testsweets/test/test_sweets_config_file_service_test.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    setUp(registerServices);
+    tearDown(() => locator.reset());
+    test(
+        "Should throw BuildError if the given app directory does not contain a .testsweets file",
+        () async {
+      final mockFileSystemService =
+          getAndRegisterFileSystemService(doesFileExist: false);
+      final configFileService = TestSweetsConfigFileServiceImplementaion();
+
+      expect(mockFileSystemService.doesFileExist(testDirectoryPath), false);
```
Comment (FilledStacks) [inline packages/testsweets/test/test_sweets_config_file_service_test.dart:24]: We can remove this expect. Unit tests mostly only test 1 unit of work at a time

File: packages/testsweets/test/test_sweets_config_file_service_test.dart [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    });
+    test(
+        "Should split any list of strings by the '=' sign to List<MapEntry<String, String>>",
+        () async {
+      TestSweetsConfigUtils testSweetsConfigUtils = TestSweetsConfigUtils();
+      var result = testSweetsConfigUtils
+          .splittingStringOnEqualSign(testsweetFileContent);
+      expect(result[0].value, '3OezzTovG9xFTE5Xw2w1');
+    });
+    test("get value from map by key", () async {
```
Comment (FilledStacks) [inline packages/testsweets/test/test_sweets_config_file_service_test.dart:46]: The test name doesn't say what we're testing. This should be in the format "When given testsweetFileContentMapEntries and ProjectId key should return `3OezzTovG9xFTE5Xw2w1` .

The test case mentioned should be inside a test group called `getStringValueFromMapByKey`. Remember each function gets a `testSuite` / `group` so we don't have to put the name of the function in the test name. The test name is specifically to tell us what parameters are give or what state it is and what we expect the outcome to be.

File: packages/testsweets/test/test_sweets_config_file_service_test.dart [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      TestSweetsConfigUtils testSweetsConfigUtils = TestSweetsConfigUtils();
+      var result = testSweetsConfigUtils.getStringValueFromMapByKey(
+          testsweetFileContentListOfMapEntries, ConfigFileKeyType.ProjectId);
+      expect(result, '3OezzTovG9xFTE5Xw2w1');
+    });
+    test("deserialize Config File by key", () async {
```
Comment (FilledStacks) [inline packages/testsweets/test/test_sweets_config_file_service_test.dart:52]: Same comment applies here that's written above.

**Review by FilledStacks — CHANGES_REQUESTED** [review]
Great PR. Some minor unit testing issues to make the tests more helpful as documentation.

**Comment by ebrahim-soliman** [comment]
Done

**Review by FilledStacks — APPROVED** [review]

**Pull request merged**